service/collector/elb.go

```diff
@@ -35,27 +34,76 @@ var (
 	)
 )
 
-func (c *Collector) collectAccountsELBs(ch chan<- prometheus.Metric, clients []aws.Clients) {
-	var wg sync.WaitGroup
+type ELBConfig struct {
+	Helper *Helper
+	Logger micrologger.Logger
+
+	InstallationName string
+}
+
+type ELBCollector struct {
+	helper *Helper
+	logger micrologger.Logger
+
+	installationName string
+}
+
+func NewELB(config ELBConfig) (*ELBCollector, error) {
+	if config.Helper == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Helper must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.InstallationName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.InstallationName must not be empty", config)
+	}
+
+	c := &ELBCollector{
+		helper: config.Helper,
+		logger: config.Logger,
+
+		installationName: config.InstallationName,
+	}
+
+	return c, nil
+}
 
-	for _, client := range clients {
-		wg.Add(1)
-		go func(awsClients aws.Clients) {
-			defer wg.Done()
-			err := c.collectELBs(ch, awsClients)
+func (c *ELBCollector) Collect(ch chan<- prometheus.Metric) error {
+	awsClientsList, err := c.helper.GetAWSClients()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var g errgroup.Group
```

Comment:
Going with `errgroup` instead of the wait group magic to have better error handling primitives.